source/particle/interpolator/bilinear_least_squares.cc

```diff
@@ -148,16 +154,88 @@ namespace aspect
                                         c[2]*(support_point[1] - approximated_cell_midpoint[1])/cell_diameter +
                                         c[3]*(support_point[0] - approximated_cell_midpoint[0])*(support_point[1] - approximated_cell_midpoint[1])/std::pow(cell_diameter,2);
 
-            cell_properties[index_positions][property_index] = interpolated_value;
-
             // Overshoot and undershoot correction of interpolated particle property.
-            if (interpolated_value > max_value_for_particle_property)
-              interpolated_value = max_value_for_particle_property;
-            else if (interpolated_value < min_value_for_particle_property)
-              interpolated_value = min_value_for_particle_property;
+            if (use_global_valued_limter)
+              {
+                interpolated_value = std::min(interpolated_value, global_maximum_particle_properties[property_index]);
+                interpolated_value = std::max(interpolated_value, global_minimum_particle_properties[property_index]);
+              }
+
+            cell_properties[index_positions][property_index] = interpolated_value;
           }
         return cell_properties;
       }
+
+      template <int dim>
+      void
+      BilinearLeastSquares<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("Interpolator");
+            {
+              prm.enter_subsection("Bilinear least squares");
+              {
+                prm.declare_entry ("Global particle property maximum",
+                                   boost::lexical_cast<std::string>(std::numeric_limits<double>::max()),
+                                   Patterns::List(Patterns::Double ()),
+                                   "The maximum global particle property values that will be used as a "
+                                   "limiter for the bilinear least squares interpolation. The number of the input "
+                                   "'Global particle property maximum' values separated by ',' has to be "
+                                   "the same as the number of the interpolated particle properties.");
```

Comment:
please remove the `interpolated` from the last line. We check whether the number of values is equal to the total number of properties, because we do not know how many will be interpolated.